pulsar/pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/GenericRecordSink.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.sink.pulsar;
+
+import com.datastax.oss.sink.RecordProcessor;
+import com.datastax.oss.sink.pulsar.record.LocalGenericRecord;
+import com.datastax.oss.sink.util.StringUtil;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.Map;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.util.Utf8;
+import org.apache.commons.lang3.SerializationException;
+import org.apache.pulsar.client.api.schema.GenericRecord;
+import org.apache.pulsar.functions.api.Record;
+import org.apache.pulsar.io.core.SinkContext;
+import org.apache.pulsar.io.core.annotations.Connector;
+import org.apache.pulsar.io.core.annotations.IOType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Connector(
+  name = "ds-cassandra",
+  type = IOType.SINK,
+  help = "PulsarSinkConnector is used for moving messages from Pulsar to Cassandra",
```

Comment:
I don't see a `PulsarSinkConnector.java` class that extends `SinkConnector` and could be an entry point for the pulsar connector. How clients will start the pulsar connector? Am I missing something?

Reply:
pulsar sinks must implement `org.apache.pulsar.io.core.Sink`

Reply:
but it should integrate with the kafka-connect framework, right? In that case, should it also implement `SinkConnector` as well?

Reply:
we're not going to integrate pulsar sink into kafka connect. what is the point of it? we are building a sink for pulsar, which would use advanced configuration and mapping features that are already implemented in kafka sink. and the pulsar sink is supposed to run within pulsar infrastructure (as a pulsar function or a standalone pulsar client).
this is why we're making efforts abstracting the core functionality and getting rid of dependency on kafka libs.